pygsti/baseobjs/label.py

```diff
@@ -201,6 +198,10 @@ class LabelTup(Label, tuple):
     acted upon by the object this label refers to.
     """
 
+    #flag used in certain Circuit subroutines
+    #Whether this is a "simple" (opaque w/a true name, from a circuit perspective) label or not.
+    is_simple= True
```

Comment:
I did consider making this an instance attribute instead of a class attribute (for presumably the same safety considerations that you have). I decided against that for two reasons. 1. the way this is set/used it really is conceptually a property of the class and not of any given instance, so semantically this makes more sense as a class attribute. 2. It added a small (but measurable) overhead to `Label` creation, and we do a *lot* of `Label` creation.

Thoughts on how best to signify this is an attribute that is meant to be read but not set? When I originally refactored this I had it 'private' with a leading underscore, but wasn't sure that signified what I wanted.

Your new thread on slots, #449 is actually quite timely here, as this is something I considered doing for the Label class (in part because it seemed it might be a way to make Label instantiation a bit faster). I ended up opting not to do so not because I thought it wasn't a good idea, but simply because I have never used slots before and was too lazy to learn how to do so properly. It sounds like it could be a real net win though in the right circumstances (and I think Labels are one one of them).

Reply:
The common way to do a read-only thing is to make it a property. I think I would be in favor of adding the leading underscore back in, having a property for the non-underscore version, and then internally using the underscore version if you want to avoid the property call overhead.

Reply:
Great, thanks for the advice. I have hybridized your suggestions, making the class attribute screaming snake case, but also adding a lower cased `is_simple` property (without a setter). I have updated the calls in `circuit.py` to use the renamed class attribute directly, but have left the rest of the half dozen or so instances in the remainder of the codebase pointing to the property (I don't anticipate the method hit mattering in those cases).

Reply:
Lol I forgot that was called screaming snake case. Sounds like the perfect convex combination of the suggestions to me :)